src/right-hand-panel/right-hand-panel-provider.js

```diff
@@ -10,7 +14,11 @@ export default function RightHandPanelProvider({ children }) {
 
     return (
         <RightHandPanelContext.Provider value={value}>
-            {children}
+            <ShowRightHandPanelContext.Provider value={show}>
```

Comment:
As far as I can see, it's not necessary to split `show` and `hide` into multiple contexts. These do not change, and can safely be in the same context. They also seem to be used together in the same component. You would probably need to `useMemo` for the context-value though. 


```javascript
// in RightHandPanelProvider
const setters = useMemo(() => {
    const show = setId
    const hide = () => setId('')
   return { show, hide } 
}, [setId])

<SetRightHandPanelContext.Provider value={setters}>


const useSetRightHandPanel = useContext(SetRightHandPanelContext)

// usage
const { show, hide } = useSetRightHandPanel()
```

It's only worth splitting the context from values that change often, and not together.